### ISSUE TITLE
Getting too many false alarms with finetuned network

### ISSUE THREAD

**Issue by sunwooOh** [weiliu89/caffe#253]
Hi, Wei. Thanks for sharing your work.

I'm trying to train text detection task(only producing bounding boxes around the text area) using the SSD network.
For the training dataset, I used SynthText in the wild([](http://www.robots.ox.ac.uk/~vgg/data/scenetext/), a large dataset with synthetic text instances) and for testing, SVT and ICDAR2013.
But still after training 50,000-100,000 iterations, it detects irrelevant objects(such as windows) and backgrounds as text, even with high confidence scores.
And getting low precision/recall/f-scores.

I plan to fix this by changing these:
- Aspect ratios of the default boxes (to more longer ones to fit text instances, like [2,4] or [2,3,5])
- Increase the weight of the location loss (partly for improved precision)
- Shrinking the network - removing conv9 layer (because the network often detects large part - half, third - of the image as text)
- Increasing confidence_threshold
- Increasing overlap_threshold and decreasing neg_pos_ratio

Do you think these will be effective?
Do you have any suggestions?

Thanks a lot.

**Comment by weiliu89** [comment]
How about if you train with real images instead of synthetic ones? I saw there are some ongoing works using SSD to detect [plate](https://github.com/hyh21521038/ssd-plate_detection) and [scene text](https://github.com/chenxinpeng/SSD_scene-text-detection). Hope it is helpful.

**Comment by sunwooOh** [comment]
Thanks @weiliu89.
Actually I've also been thinking that synthetic text data has some major disadvantages.
And thank you also for the links.
I'll try training with both real world/synthetic data.

**Comment by weiliu89** [comment]
As a general suggestion, I would first make suer the data and annotation are prepared correct. You can try to use examples/ssd.ipynb to check and make sure everything is right.

**Comment by sunwooOh** [comment]
I check that with similar code as ssd.ipynb, thanks :)

**Comment by sunwooOh** [comment]
Update:
Dataset seems to have been the problem.
I made the dataset again, for some different reason, but after that my network's getting trained fine!
Thanks Wei :+1:
